docs/standard/security/cryptographic-signatures.md

```diff
@@ -23,149 +23,156 @@ ms.assetid: aa87cb7f-e608-4a81-948b-c9b8a1225783
+
+The classes in this namespace follow the [XML-Signature Syntax and Processing recommendation](https://www.w3.org/TR/xmldsig-core/) from the World Wide Web Consortium.
+
+[Back to top](#top)
+
+<a name="verify"></a>
+
+## Verifying Signatures
+
+To verify that data was signed by a particular party, you must have the following information:
+
+- The public key of the party that signed the data.
+
+- The digital signature.
+
+- The data that was signed.
+
+- The hash algorithm used by the signer.
+
+To verify a signature signed by the <xref:System.Security.Cryptography.RSAPKCS1SignatureFormatter> class, use the <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class. The <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> class must be supplied the public key of the signer. You will need the values of the modulus and the exponent to specify the public key. (The party that generated the public/private key pair should provide these values.) First create an <xref:System.Security.Cryptography.RSACryptoServiceProvider> object to hold the public key that will verify the signature, and then initialize an <xref:System.Security.Cryptography.RSAParameters> structure to the modulus and exponent values that specify the public key.
+
+The following code shows the creation of an <xref:System.Security.Cryptography.RSAParameters> structure. The `Modulus` property is set to the value of a byte array called `modulusData` and the `Exponent` property is set to the value of a byte array called `exponentData`.
+
+```vb
+Dim rsaKeyInfo As RSAParameters
+rsaKeyInfo.Modulus = modulusData
+rsaKeyInfo.Exponent = exponentData
+```
+
+```csharp
+RSAParameters rsaKeyInfo;
+rsaKeyInfo.Modulus = modulusData;
+rsaKeyInfo.Exponent = exponentData;
+```
+
+After you have created the <xref:System.Security.Cryptography.RSAParameters> object, you can initialize a new instance of the <xref:System.Security.Cryptography.RSACryptoServiceProvider> class to the values specified in <xref:System.Security.Cryptography.RSAParameters>. The <xref:System.Security.Cryptography.RSACryptoServiceProvider> is, in turn, passed to the constructor of an <xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter> to transfer the key.
+
+The following example illustrates this process. In this example, `hashValue` and `signedHashValue` are arrays of bytes provided by a remote party. The remote party has signed the `hashValue` using the SHA1 algorithm, producing the digital signature `signedHashValue`. The
+
+<xref:System.Security.Cryptography.RSAPKCS1SignatureDeformatter.VerifySignature%2A?displayProperty=nameWithType> method verifies that the digital signature is valid and was used to sign the `hashValue`.
```

Comment:
This line should be pulled up to the `The` of the previous sentence.

Reply:
Done